<!-- amcl launch -->
<launch >
	<!-- arguments -->
	<arg name="map_yaml_file" default="$(find robot_sim)/../../maps/nic_garage/ogm.yaml" />

	<!-- tf -->
	<node name="world_to_map" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 world map 100" />
	<node name="amcl_frame_to_base_link" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 amcl_frame base_link 100" />

	<!-- map server -->
	<node name="amcl_map_server" pkg="map_server" type="map_server" args="$(arg map_yaml_file)" >
		<remap from="/map" to="/amcl_map" />
		<remap from="/map_metadata" to="/amcl_map_metadata" />
	</node >

	<!-- amcl localizer -->
	<node name="amcl" pkg="localizer" type="amcl" output="screen" >
		<param name="/map_frame" value="/world" type="string" />
		<param name="/laser_frame" value="/laser" type="string" />
		<param name="/base_link_frame" value="/base_link" type="string" />
		<param name="/input_map_topic_name" value="/amcl_map" type="string" />
		<param name="/input_odom_topic_name" value="/odom" type="string" />
		<param name="/input_scan_topic_name" value="/scan" type="string" />
		<param name="/min_particle_num" value="200" type="int" />
		<param name="/max_particle_num" value="500" type="int" />
		<param name="/resample_threshold" value="0.5" type="double" />
		<param name="/scan_step" value="10" type="double" />
		<param name="/max_dist_to_obstacle" value="0.5" type="double" />
		<param name="/alpha_slow" value="0.0001" type="double" />
		<param name="/alpha_fast" value="0.1" type="double" />
		<param name="/update_dist" value="0.2" type="double" />
		<param name="/update_yaw" value="2.0" type="double" />
		<param name="/update_time" value="5.0" type="double" />
		<param name="/odom_noise_dist_dist" value="1.5" type="double" />
		<param name="/odom_noise_dist_head" value="0.9" type="double" />
		<param name="/odom_noise_head_dist" value="0.9" type="double" />
		<param name="/odom_noise_head_head" value="2.5" type="double" />
		<param name="/start_x" value="0.0" type="double" />
		<param name="/start_y" value="0.0" type="double" />
		<param name="/start_yaw" value="0.0" type="double" />
		<param name="/initial_cov_xx" value="0.2" type="double" />
		<param name="/initial_cov_yy" value="0.2" type="double" />
		<param name="/initial_cov_yawyaw" value="1.0" type="double" />
		<param name="/pose_publish_hz" value="20.0" type="double" />
		<param name="/use_kld_sampling" value="true" type="bool" />
		<param name="/use_test_range_measurement" value="true" type="bool" />
		<param name="/dynamic_scan_point_threshold" value="0.9" type="double" />
	</node >
</launch >
